monai/handlers/checkpoint_loader.py

```diff
@@ -80,5 +80,11 @@ def __call__(self, engine: Engine) -> None:
         """
         checkpoint = torch.load(self.load_path, map_location=self.map_location)
 
+        # save current max epochs setting in the engine, don't overwrite it if larger than max_epochs in checkpoint
+        prior_max_epochs = engine.state.max_epochs
         Checkpoint.load_objects(to_load=self.load_dict, checkpoint=checkpoint)
+        if engine.state.epoch > prior_max_epochs:
+            raise ValueError("current max epochs setting is smaller than epoch count in the checkpoint.")
```

Comment:
Maybe a nit, a small test entering this line could increase your codecov.

Reply:
Hi @vfdev-5 ,

Thanks for your review.
I added more unit tests to cover it.

Thanks.